torch/_functorch/partitioners.py

```diff
@@ -1404,10 +1403,11 @@ def _optimize_runtime_with_given_memory(
     elif SOLVER == "dp":
         return dp_knapsack(memory, runtimes, max_memory)
     elif callable(SOLVER):
-        saved_node_idx, recomp_node_idx = SOLVER(
-            memory, joint_graph, max_memory, node_info, all_recomputable_banned_nodes
+        solver_instance = SOLVER()
```

Comment:
Updating how the callable should be defined. Here is an example:

```
class ExampleCustomKnapsackSolver:
    def __init__(self) -> None:
        pass

    def __call__(
        self,
        joint_graph: torch.fx.Graph,
        all_recomputable_banned_nodes: List[torch.fx.Node],
        memory: List[float],
        runtimes: List[float],
        max_memory: float,
    ) -> Tuple[float, List[int], List[int]]:
      ...
      return expected_runtime, saved_node_idx, recomp_node_idx
```

Reply:
This API seems unnessarily complex.  You should be able to add a wrapper and still use the old API:
```py
def call_solver(memory, joint_graph, max_memory, node_info, all_recomputable_banned_nodes)
   tmp = ExampleCustomKnapsackSolver()
   return tmp(memory, joint_graph, max_memory, node_info, all_recomputable_banned_nodes)

SOLVER = call_solver
```

Though adding the extra `expected_runtime` return which seems fine.